### PR TITLE
Convert the Android source code to Gradle.

### PR DESCRIPTION
**Pull request by sigmabeta**
The Android project was converted to use the Gradle build system before there were complete tools for doing so. As a result, build.gradle contains a lot of scripting pointing to various directories, breaking compatibility with some UI libraries, and creating unnecessary complexity for new developers.

Although almost every file in the Android directory is moved around by this commit, it should not make any meaningful changes to functionality.

### CODE REVIEW

**Comment by sigmabeta** [comment]
An additional SDK component needs to be installed on buildbot for this build to succeed.

Building this PR currently fails because as part of this conversion, I'm no longer including the Android Support Library as a JAR file in the repo - rather, I've instructed the gradle script to pull it down (if necessary) at build-time (lines 65+66 of build.gradle). 

This approach allows us to add or remove libraries by adding or removing the respective line in build.gradle, rather than having to include the library in a commit.

However, the support library is unique in that it can't be found in the MavenCentral or JCenter repositories (what Gradle pulls from by default): it uses a special Google-backed repository, access to which is gated by installing the appropriate SDK component.

tl;dr someone needs to ssh into buildbot and run:
```
<path-to-android-sdk>/tools/android update sdk --no-ui --filter extra
```

**Comment by sigmabeta** [comment]
Paging @Parlane as he is apparently the person who handles this type of stuff?

**Comment by Sonicadvance1** [comment]
@dolphin-emu-bot rebuild
